heidrun/nara_json.rb

```diff
@@ -0,0 +1,575 @@
+def nara_catalog_uri(id)
+  "http://catalog.archives.gov/id/#{id.node}"
+end
+
+# See object and preview mappings
+def make_obj_uri(obj)
+  URI::escape(obj.node['@url'])  # Can have space characters
+end
+def make_obj_dcformat(obj)
+  obj.node['@mime']
+end
+def make_preview_uri(obj)
+  URI::escape(obj.node['@url'])  # Can also have space characters
+end
+def make_preview_dcformat(obj)
+  obj.node['@mime']
+end
+
+# Return a string suitable for sourceResource.contributor or
+# sourceResource.publisher.
+#
+# @see #make_contributor
+# @see #make_publisher
+# @param contributor_array [Krikri::JsonParser::Value]
+# @return [String, RDF::Literal]
+def contributor_term_name(contributor_array)
+  node = contributor_array.node
+  contributors = node.fetch('organizationalContributor',
+                            node['personalContributor'])
+  contributors = [contributors] unless contributors.is_a? Array
+
+  yield(contributors)
+
+  return contributors.first['contributor']['termName'] \
+    unless contributors.empty?
+
+  # FIXME:  It's not clear how to indicate the concept of "nothing" to
```

Comment:
I'm not sure if this is the right approach, but I'm not sure what the best way to tackle it would be. A couple of options: 

* You could have something in the mapping reject any `nil` values. 
* You could add similar logic in `#make_contributor`, perhaps in the `contributors.select! ...` block. 
* The mapping DSL might need to have additional logic to understand what it means when it receives `nil` for a given property; e.g. if there are no other properties with URI or Literal values, nor does it have an associated URI, it is treated as an empty object. This might be a bigger conversation for when @no-reply returns from vacation.